eos-paygd/main.c

```diff
@@ -583,11 +574,22 @@ main (int   argc,
       eospayg_efi_root_pivot ();
     }
 
-  /* Try to log to a file in /var, in addition to the journal, so we have
-   * persistent logs even on systems with fragile storage.
+  /* Use /bin/mkdir instead of mkdir() to ensure the mode is unaffected by
+   * the process's umask.
    */
-  if (open_log_file (&log_io))
-    g_log_set_writer_func (log_writer, log_io, NULL);
+  system_ret = system ("/bin/mkdir -p --mode=755 " LOGFILE_DIRNAME);
```

Comment:
Does this run before or after the real `/var` is available?

Reply:
Yes, `/var` is mounted by `ostree-prepare-root.service`, ordered `Before=initrd-root-fs.target`, whereas `eos-paygd-2.service` is ordered `After=initrd-root-fs.target`.

At this point in the code paygd has chroot'ed itself to `/sysroot` and was able to connect to D-Bus's system bus through a busy loop, so we now the real root is in a good state.

I think we can actually create the logdir earlier by pointing it to `/sysroot/var` when running from the initramfs, but I need to try it out.